tests/unauthorized/filesTest.js

```diff
@@ -1,34 +1,129 @@
+    'Access-Control-Allow-Methods': 'GET,OPTIONS,POST,PUT,DELETE,MKCOL,PROPFIND,PATCH,PROPPATCH,REPORT,HEAD,COPY,MOVE'
+  }
+  const unauthorizedResponse = {
+    status: 401,
+    headers: webdavFilesResponseHeader,
+    body: incorrectAuthorizationXmlResponseBody
+  }
+  const webdavUrl = Pact.Matchers.regex({
+    matcher: '.*\\/remote\\.php\\/webdav\\/.*',
+    generate: `/remote.php/webdav/${config.testFolder}`
+  })
 
   beforeAll(function (done) {
-    Promise.all(setGeneralInteractions(provider)).then(done, done.fail)
+    const promises = []
+    promises.push(setGeneralInteractions(provider))
+    promises.push(provider.addInteraction({
+      uponReceiving: 'a request to get file contents with invalid authentication',
+      withRequest: {
+        method: 'GET',
+        path: webdavUrl,
+        headers: requestHeaderWithInvalidAuth
+      },
+      willRespondWith: unauthorizedResponse
+    }))
+    promises.push(provider.addInteraction({
+      uponReceiving: 'a request to put file contents with invalid authentication',
+      withRequest: {
+        method: 'PUT',
+        path: webdavUrl,
+        headers: requestHeaderWithInvalidAuth
+      },
+      willRespondWith: unauthorizedResponse
+    }))
+    promises.push(provider.addInteraction({
+      uponReceiving: 'a request to delete file contents with invalid authentication',
+      withRequest: {
+        method: 'DELETE',
+        path: webdavUrl,
+        headers: requestHeaderWithInvalidAuth
+      },
+      willRespondWith: unauthorizedResponse
+    }))
+    promises.push(provider.addInteraction({
+      uponReceiving: 'a request to create a directory with invalid authentication',
+      withRequest: {
+        method: 'MKCOL',
+        path: webdavUrl,
+        headers: {
+          authorization: invalidAuthHeader,
+          Origin: origin
+        }
+      },
+      willRespondWith: {
+        status: 401,
+        headers: webdavFilesResponseHeader,
+        body: incorrectAuthorizationXmlResponseBody
+      }
+    }))
+    promises.push(provider.addInteraction({
+      uponReceiving: 'a request to list files with invalid authentication',
+      withRequest: {
+        method: 'PROPFIND',
+        path: webdavUrl,
+        headers: requestHeaderWithInvalidAuth
+      },
+      willRespondWith: unauthorizedResponse
+    }))
+    promises.push(provider.addInteraction({
+      uponReceiving: 'a request to move file with invalid authentication',
+      withRequest: {
+        method: 'MOVE',
+        path: webdavUrl,
+        headers: requestHeaderWithInvalidAuth
+      },
+      willRespondWith: unauthorizedResponse
+    }))
+    promises.push(provider.addInteraction({
+      uponReceiving: 'a request to copy a file with invalid authentication',
+      withRequest: {
+        method: 'COPY',
+        path: webdavUrl,
+        headers: requestHeaderWithInvalidAuth
+      },
+      willRespondWith: unauthorizedResponse
+    }))
+
```

Comment:
all those could be summarised in one single interaction using matchers https://github.com/pact-foundation/pact-js/#matching

Reply:
matchers seem to be not working for `method` so I've done refactoring with a for loop.